extensions/amp-form/0.1/amp-form.js

```diff
@@ -124,20 +126,25 @@ export class AmpForm {
 
   /**
    * Adds proper classes for the state passed.
-   * @param {string} state
+   * @param {string} newState
    * @private
    */
-  setState_(state) {
-    this.form_.classList.remove(`amp-form-${this.state_}`);
-    this.form_.classList.add(`amp-form-${state}`);
-    this.state_ = state;
+  setState_(newState) {
+    const previousState = this.state_;
+    this.form_.classList.remove(`amp-form-${previousState}`);
+    this.form_.classList.add(`amp-form-${newState}`);
+    this.state_ = newState;
     this.submitButtons_.forEach(button => {
-      if (state == FormState_.SUBMITTING) {
+      if (newState == FormState_.SUBMITTING) {
         button.setAttribute('disabled', '');
       } else {
         button.removeAttribute('disabled');
       }
     });
+
+    if (newState == FormState_.SUBMITTING) {
```

Comment:
I think this'd be more appropriate before setting the state to submitting (before calling `#setState`)